akka-http-core/src/main/scala/akka/http/scaladsl/Http.scala

```diff
@@ -43,6 +43,8 @@ class HttpExt(private val config: Config)(implicit val system: ActorSystem) exte
 
   override val sslConfig = AkkaSSLConfig(system)
 
+  private[this] val defaultConnectionPoolSettings = ConnectionPoolSettings(system)
```

Comment:
While this is a good change by itself, we also need to fix equality for ConnectionPoolSettings.